### PR TITLE
Allow disabling the dedicated gateways created by cloud prepare

### PR DESCRIPTION
**Pull request by mangelajo**
Adds --disable-gateways parameter for:

   `subctl cloud prepare aws`

This functionality is to be used with the LoadBalancer support which
doesn't require dedicated gateways, (and doesn't work with them
right now if they are on the public subnet).

Fixes-Issue: https://github.com/submariner-io/cloud-prepare/issues/49
Related-Issue: https://github.com/submariner-io/submariner/issues/1071
Depends on https://github.com/submariner-io/cloud-prepare/pull/53
Signed-off-by: Miguel Angel Ajo <majopela@redhat.com>

### CODE REVIEW

File: pkg/subctl/cmd/cloud/prepare/aws.go [hunk 1]
```diff
@@ -26,26 +26,40 @@ import (
...
 // NewCommand returns a new cobra.Command used to prepare a cloud infrastructure
 func newAWSPrepareCommand() *cobra.Command {
 	cmd := &cobra.Command{
-		Use:   "aws",
-		Short: "Prepare an AWS cloud",
-		Long:  "This command prepares an AWS based cloud for Submariner installation.",
-		Run:   prepareAws,
+		Use:    "aws",
+		Short:  "Prepare an AWS cloud",
+		Long:   "This command prepares an AWS based cloud for Submariner installation.",
```
Comment (dfarrell07) [inline pkg/subctl/cmd/cloud/prepare/aws.go]:
```suggestion
		Long:   "This command prepares an AWS-based cloud for Submariner installation.",
```

File: pkg/subctl/cmd/cloud/prepare/aws.go [hunk 2]
```diff
@@ -26,26 +26,40 @@ import (
...
+		Run:    prepareAws,
+		PreRun: validateArgsAws,
 	}
 
 	aws.AddAWSFlags(cmd)
 	cmd.Flags().StringVar(&gwInstanceType, "gateway-instance", "m5n.large", "Type of gateways instance machine")
-	cmd.Flags().IntVar(&gateways, "gateways", 1, "Amount of gateways to prepare (0 = gateway per public subnet)")
-
+	cmd.Flags().IntVar(&gateways, "gateways", DefaultDedicatedGateways, "Amount of gateways to prepare (0 = gateway per public subnet)")
```
Comment (dfarrell07) [inline pkg/subctl/cmd/cloud/prepare/aws.go]:
```suggestion
	cmd.Flags().IntVar(&gateways, "gateways", DefaultDedicatedGateways, "Number of gateways to prepare (0 = gateway per public subnet)")
```

File: pkg/subctl/cmd/cloud/prepare/aws.go [hunk 3]
```diff
@@ -26,26 +26,40 @@ import (
...
+	cmd.Flags().BoolVarP(&disableDedicatedGateways, "disable-gateways", "d", false, "Setup no dedicated gateways "+
```
Comment (dfarrell07) [inline pkg/subctl/cmd/cloud/prepare/aws.go]:
```suggestion
	cmd.Flags().BoolVarP(&disableDedicatedGateways, "disable-gateways", "d", false, "Set up no dedicated gateways "+
```

**Comment by mangelajo** [comment]
Thanks @dfarrell07, handling.

**Review by tpantelis — APPROVED** [review]

**Review by dfarrell07 — APPROVED** [review]

**Review by skitt — APPROVED** [review]

**Pull request merged**